terminal-view/src/main/java/com/termux/view/TerminalRenderer.java

```diff
@@ -117,11 +117,10 @@ public final void render(TerminalEmulator mEmulator, Canvas canvas, int topRow,
                     } else {
                         final int columnWidthSinceLastRun = column - lastRunStartColumn;
                         final int charsSinceLastRun = currentCharIndex - lastRunStartIndex;
-                        int cursorColor = (lastRunInsideCursor || lastRunInsideSelection) ? mEmulator.mColors.mCurrentColors[TextStyle.COLOR_INDEX_CURSOR] : 0;
-                        int cursorStyle = lastRunInsideSelection ? TerminalEmulator.CURSOR_STYLE_BLOCK : cursorShape;
+                        int cursorColor = lastRunInsideCursor ? mEmulator.mColors.mCurrentColors[TextStyle.COLOR_INDEX_CURSOR] : 0;
                         drawTextRun(canvas, line, palette, heightOffset, lastRunStartColumn, columnWidthSinceLastRun,
                             lastRunStartIndex, charsSinceLastRun, measuredWidthForRun,
-                            cursorColor, cursorStyle, lastRunStyle, reverseVideo);
+                            cursorColor, cursorShape, lastRunStyle, reverseVideo || lastRunInsideSelection);
```

Comment:
Aren't you effectively reverting your last PR (#1627) here?

Reply:
This reverts the cursor style part but it still depends on the `insideSelection` logic from #1627. I chose to submit this separately since I wanted to fix the existing behavior first before changing anything.

Reply:
Ah, okay, and since you're drawing the selection by reversing video instead of by using the cursor it isn't necessary anymore.